components/script/dom/htmliframeelement.rs

```diff
@@ -360,6 +373,41 @@ impl HTMLIFrameElement {
         let window = window_from_node(self);
         window.reflow(ReflowGoal::Full, ReflowReason::IFrameLoadEvent);
     }
+
+    fn get_window_size(&self) -> WindowSizeData {
+        // iframes are replaced elements, so they have a default size of 300x150, according to
+        // https://www.w3.org/TR/CSS2/visudet.html#inline-replaced-width
+        let window = window_from_node(self);
```

Comment:
This looks wrong. This looks at the iframes `width` and `height` attributes, but what if I do `<iframe style="height: 400px">`? What if I use a CSS rule instead?

Reply:
Huh, good point, but which one takes precedence?

Reply:
Depends on whether the rule is `!important` vs. the style attribute, etc... In general you should check the layout size of the fragment here. But this looks like a workaround for the viewport not being set at the right time...

Reply:
Mind if I file a followup to 8nvestigate this? The scope is actually quite big here, e.g. we need to account for `display: none`.

Reply:
I sort of do... This approach of knowing the viewport size before creating the bc is kinda broken and is going to be reverted... Do you know what makes us not have the viewport size at the time?

Reply:
I don't know, from my code reading it looks like it was just never set at all, and the reason why this comes up now is most likely because we actually return a legitimate BC right now instead of it just being None.